packages/cli/src/commands/multi-chain/deploy.ts

```diff
@@ -0,0 +1,200 @@
+              true
+            );
+          } else {
+            indexerVersions[validator.chainId] = indexerAvailableVersions[0];
+          }
+        } catch (e) {
+          throw new Error(chalk.bgRedBright('Indexer version is required'));
+        }
+      }
+
+      indexerVersions[validator.chainId] = addV(indexerVersions[validator.chainId]);
+
+
+      if (!endpoints[validator.chainId]) {
+        if (flags.useDefaults) {
+          throw new Error(chalk.red('Please ensure a endpoint valid is passed using --endpoint flag with syntax chainId:rpc_endpoint,chainId2:rpc_endpoint2...'));
+        }
+
+        endpoints[validator.chainId] = await promptWithDefaultValues(cli, `Enter endpoint for ${multichainProjectPath}`, undefined, null, true);
+      }
+
+
+      if (!dictionaries[validator.chainId])
+      {
+        const validateDictEndpoint = processEndpoints(await dictionaryEndpoints(ROOT_API_URL_PROD), validator.chainId);
+        if (!flags.useDefaults && !validateDictEndpoint) {
+          dictionaries[validator.chainId] = await promptWithDefaultValues(cli, `Enter dictionary for ${multichainProjectPath}`, validateDictEndpoint, null, false);
+        } else {
+          dictionaries[validator.chainId] = validateDictEndpoint;
+        }
+      }
+
+
+      chains.push(
+        generateDeploymentChain({
+          cid: String(multichainProjectCid),
+          dictEndpoint: String(dictionaries[validator.chainId]),
+          endpoint: [endpoints[validator.chainId]],
+          flags: flags as unknown as deploymentFlagsInterface,
+          indexerImageVersion: String(indexerVersions[validator.chainId])
+        })
+      );
+    }
+
+    this.log('Deploying SubQuery multi-chain project to Hosted Service');
+
+    let deploymentOutput = await executeProjectDeployment({
+      authToken: authToken,
+      chains: chains,
+      flags: flags as unknown as deploymentFlagsInterface,
+      ipfsCID: String(ipfsCID),
+      org: String(org),
+      projectInfo: projectInfo,
+      projectName: String(projectName),
+      queryVersion: queryVersion
+    });
+
+    if (!deploymentOutput) {
+      this.log(`Project: ${projectName} has been re-deployed`);
+    } else {
+      this.log(`Project: ${deploymentOutput.projectKey}
+      \nStatus: ${chalk.blue(deploymentOutput.status)}
+      \nDeploymentID: ${deploymentOutput.id}
+      \nDeployment Type: ${deploymentOutput.type}
+      \nIndexer version: ${deploymentOutput.indexerImage}
+      \nQuery version: ${deploymentOutput.queryImage}
+      \nEndpoint: ${deploymentOutput.endpoint}
+      \nDictionary Endpoint: ${deploymentOutput.dictEndpoint}
+      \nQuery URL: ${deploymentOutput.queryUrl}
+      \nProject URL: ${BASE_PROJECT_URL}/project/${deploymentOutput.projectKey}
+      \nAdvanced Settings for Query: ${JSON.stringify(deploymentOutput.configuration.config.query)}
+      \nAdvanced Settings for Indexer: ${JSON.stringify(deploymentOutput.configuration.config.indexer)}
+      `);
+    }
```

Comment:
This can be shared with the non-multichain code

Reply:
moved.